File: sandbox/python_go/send_json.py
import socket
import json

target_IP = "127.0.0.1"
target_PORT = 16384

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

print("JSON mode")
print("send 'exit' as name to finish")

while True:
    name_input = input("Enter the name >:")
    if name_input == "exit":
        payout_input = "0"
    else:
        payout_input = input("Enter the payout >:")

    if payout_input == "cancel":
        continue

    data = {
        "name" : name_input,
        "payout": int(payout_input)
    }

    json_data = json.dumps(data)

    sock.sendto(json_data.encode('utf-8'), (target_IP, target_PORT))

    if name_input == "exit":
        print("detected exit command")
        break

sock.close()

File: sandbox/python_go/send_json.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"net"
	"os"
	"strconv"
)

type gamedata struct {
	Name   string `json:"name"`
	Payout int    `json:"payout"`
}

func main() {
	conn, err := net.Dial("udp", "127.0.0.1:16384")
	if err != nil {
		fmt.Println(err)
		return
	}
	defer conn.Close()

	fmt.Println("JSON mode")
	fmt.Println("send 'exit' as name to finish")

	buffer := bufio.NewScanner(os.Stdin)

	for {

		fmt.Print("Enter the name >:")

		if !buffer.Scan() {
			break
		}

		nameInput := buffer.Text()

		if nameInput == "exit" {
			exitData := gamedata{Name: "exit", Payout: 0}
			exitData_json, _ := json.Marshal(exitData)
			conn.Write(exitData_json)
			fmt.Println("exit command detected")
			break
		}

		fmt.Print("Enter the payout >:")

		if !buffer.Scan() {
			break
		}

		payoutInput := buffer.Text()
		payoutInt, err := strconv.Atoi(payoutInput)

		if err != nil {
			fmt.Println("error do not enter string as payout")
			continue
		}

		data := gamedata{
			Name:   nameInput,
			Payout: payoutInt,
		}

		json_data, err := json.Marshal(data)
		if err != nil {
			fmt.Println("json error", err)
			continue
		}

		conn.Write(json_data)
	}
}
